tests/src/core/testqgsdatasourceuri.cpp

```diff
@@ -524,5 +525,11 @@ void TestQgsDataSourceUri::checkParameterKeys()
   QVERIFY( uri.parameterKeys().contains( QLatin1String( "bar" ) ) );
 }
 
+void TestQgsDataSourceUri::checkRemovePassword()
+{
+  const QString uri = QgsDataSourceUri::removePassword( QStringLiteral( "postgresql://user:password@127.0.0.1:5432?dbname=test" ) );
+  QCOMPARE( uri, QStringLiteral( "postgresql://user@127.0.0.1:5432?dbname=test" ) );
```

Comment:
Side request, I would be nice to have an option in `QgsDataSourceUri::removePassword` to actually not "remove" the password but replacing it by `XXXXXXXXX`. I got one day confused about the logs itself in another area of QGIS server about not knowing if the password was really missing from the datasource or just removed by this function :)

Reply:
I added a `hide` parameter to `removePassword()` static method